uxarray/utils/helpers.py

```diff
@@ -673,3 +678,114 @@ def close_face_nodes(Mesh2_face_nodes, nMesh2_face, nMaxMesh2_face_nodes):
     np.put(closed.ravel(), first_fv_idx_1d, first_node_value)
 
     return closed
+
+
+def point_within_GCR(pt, gcr_cart):
+    """Check if a point is on a given Great Circle Arc. The anti-meridian case
+    is already considered.
+
+    Parameters
+    ----------
+    pt : np.ndarray of float
+        Cartesian coordinates of the point
+    gcr_cart : np.ndarray of shape (2, 3), (np.float or gmpy2.mpfr)
+        Cartesian coordinates of the GCR
+
+    Returns
+    -------
+    bool
+        True if the point is on the GCR (Lies between the two endpoints), False otherwise
+    """
+    # Convert the cartesian coordinates to lonlat coordinates, the node_xyz_to_lonlat_rad is already overloaded
+    # with gmpy2 data type
+    pt_lonlat = node_xyz_to_lonlat_rad(pt)
+    GCRv0_lonlat = node_xyz_to_lonlat_rad(gcr_cart[0])
+    GCRv1_lonlat = node_xyz_to_lonlat_rad(gcr_cart[1])
```

Comment:
Should we consider allowing the inputted point and GCR to be in lat/lon, with an argument specifying whether the input is either Cartesian or lat/lon?

```Python
def point_within_GCR(pt, gcr, latlon=True):
```

```Python
if not latlon:
    pt = node_xyz_to_lonlat_rad(pt)
    GCRv0 = node_xyz_to_lonlat_rad(gcr_cart[0])
    GCRv1 = node_xyz_to_lonlat_rad(gcr_cart[1])
else:
    GCRv0 = gcr_cart[0]
    GCRv1 = gcr_cart[1]
```

Additionally, I would also suggest working in degrees instead of radians, since our coordinates throughout are typically in lat/lon degrees.

Reply:
> All internal calculations use the Cartesian coordinates for computational precision and algorithms robustness. And we want to reduce the number of conversion calls to avoid accumulated errors as much as possible.

This makes sense in the scope of your work and I don't have any concerns about it. However, it would be useful to not limit this helper function to only allowing inputs in Cartesian coordinates, since the lines above would _always_ attempt to convert them into lat/lon. 

> One thing that should be noticed is that the radian and degree conversion is tough to maintain the precision in general, no matter what package is used (including numpy). So it will be better to only do this conversion at the beginning or at the end.

Our coordinates (i.e. `Mesh2_node_x`) are expected to be in degrees, which is why I suggested using degrees for the check. 
Can you explain the issues that you've run into when converting between radians and degrees?

Reply:
That's fair, we can keep the function the way it is right now and update it to support lat/lon down the line if we need to use it for other applications.